### PR TITLE
Add tests for weather function, explicit function calls

### PR DESCRIPTION
**Pull request by gmyenni**
Also remove FullPath from weather function.

### CODE REVIEW

**Comment by gmyenni** [comment]
All the tests pass for me locally, so I'm not sure what's going on here.

**Comment by ethanwhite** [comment]
This originally passed for me locally, but after I removed my local copy of `~/PortalData/` it failed with the same error.

I'm testing in the same way that Travis does, which is to go the directory above your repo (i.e., the directory that the repo directory is in) and run:

```
R CMD build PortalDataSummaries
R CMD check PortalDataSummaries_0.0.0.9000.tar.gz
```

Happy to help with debugging if it's useful at some point.

**Comment by gmyenni** [comment]
I fixed the issue with the weather checks, but in the process it rewrote the NAMESPACE file incorrectly. I'm not sure why. (I started changing what we need in the namespace as was mentioned [here](https://github.com/weecology/PortalDataSummaries/issues/25), but I assume that's still why it's not passing?)

File: R/Weather.R [hunk 1]
```diff
@@ -1,26 +1,7 @@
...
-#'
-#' @param ReferencePath a path to join with current working directory
-#' @param BasePath Current working directory else path given
-#'
-#' @return
-#' @export
-#' @examples
-#' FullPath('PortalData/Rodents/Portal_rodent.csv')
-#' FullPath('PortalData/Rodents/Portal_rodent.csv', '~')
-FullPath <- function( ReferencePath, BasePath=getwd()){
```
Comment (ethanwhite) [inline R/Weather.R]: Since we're removing `FullPath` here it seems like we should need to get it elsewhere when it is used on on the lines that load data.

Comment (ethanwhite) [inline R/Weather.R]: Never mind, it's coming from `GetData`, sorry. I have to admit I still don't fully understand how to think about namespacing within packages in R.

**Comment by ethanwhite** [comment]
Weird, the tests as a whole were passing for the PR, but now appear to be failing after the merge...

**Comment by gmyenni** [comment]
Yeah, something is going on here I don't understand. Everything passes for me and works fine.

**Comment by ethanwhite** [comment]
I can confirm that we have a test that is failing locally:

```
> devtools::test('PortalDataSummaries')
Loading PortalDataSummaries
Testing PortalDataSummaries
checks data_processing functions: ............
|===================================================== |100% ~0 s remaining     1W.
checks weather summary output: ..W.....W...

Warnings -----------------------------------------------------------------------
1. data generated by default setting is same (@test-regression.R#7) - joining factor and character vector, coercing into character vector

2. data generated by level=treatment, length=long-term is same (@test-regression.R#15) - joining factor and character vector, coercing into character vector

3. data generated by level=plot, time=newmoon, type=granivore, shape=flat 
          is same (@test-regression.R#22) - joining factor and character vector, coercing into character vector

4. data generated by unknowns=T, incomplete=T is same (@test-regression.R#29) - joining character vector and factor, coercing into character vector

5. Monthly option returns 7 columns (@test-weather.R#12) - NAs introduced by coercion

6. Monthly temperatures ok (@test-weather.R#26) - NAs introduced by coercion

Failed -------------------------------------------------------------------------
1. Failure: data generated by level=plot, time=newmoon, type=granivore, shape=flat 
          is same (@test-regression.R#25) 
digest(data) not identical to "80a297349b34d9eef00da1a6eeb0ec3d".
1/1 mismatches
x[1]: "487361feea3d07b7679c893d8bc0abdb"
y[1]: "80a297349b34d9eef00da1a6eeb0ec3d"


DONE ===========================================================================
Warning messages:
1: character(0) 
2: character(0) 
3: character(0) 
4: character(0) 
```

It's a hash mismatch in the third regression test.

The two most likely options for what is going on are:
1. A recent change in the data changed something about the output. This could have been the fix to the duplicates that was reported and merged a week or two ago. If that's true then @skmorgane just needs to recheck everything and update the hash if she's happy.
2. There is an OS difference in the hash for some reason (e.g., line endings) so it's failing on Linux (me and Travis) but not for you.

**Pull request merged**